browser-extension/mv2/src/background/background.js

```diff
@@ -1385,14 +1389,15 @@ BG.Methods.saveTestRuleResult = (payload, senderTab) => {
   });
 };
 
-BG.Methods.startRecordingExplicitly = (tabId, showWidget = true) => {
+BG.Methods.startRecordingExplicitly = async (tabId, showWidget = true, url) => {
```

Comment:
```suggestion
BG.Methods.startRecordingExplicitly = async (tab, showWidget = true) => {
```
Lets pass the `Tab` object here